src/relax/ir/dataflow_matcher.cc

```diff
@@ -1140,34 +1071,173 @@ class PatternRewriter : ExprMutator {
+ * matches with the output of a user-provided rewriter function.
+ */
+class ExprPatternRewriter : ExprMutator {
+ public:
+  using ExprMutator::VisitBindingBlock_;
+  using ExprMutator::VisitExpr_;
+
+  ExprPatternRewriter(DFPattern pat,
+                      TypedPackedFunc<Expr(Expr, Map<DFPattern, Expr>)> rewriter_func)
+      : pattern_(pat), rewriter_func_(rewriter_func) {}
+
+  template <typename PatternType>
+  static Function Run(PatternType pat,
+                      TypedPackedFunc<Expr(Expr, Map<DFPattern, Expr>)> rewriter_func,
+                      Function func) {
+    ExprPatternRewriter rewriter(pat, rewriter_func);
+    func = Downcast<Function>(rewriter(func));
+    func = Downcast<Function>(RemoveAllUnused(func));
+    return func;
+  }
+
+  Expr VisitExpr_(const SeqExprNode* seq) override {
+    auto cache = bindings_;
+    SeqExpr prev = GetRef<SeqExpr>(seq);
+
+    StructuralEqual struct_equal;
+
+    while (true) {
+      SeqExpr next = Downcast<SeqExpr>(builder_->Normalize(ExprMutator::VisitExpr_(prev.get())));
+      if (struct_equal(prev, next)) {
+        return std::move(next);
+      }
+
+      // Canonicalization may result in two previously-different
+      // expressions being recognized as identical.  Elimination of
+      // common subexpressions may result in trival var-to-var
+      // bindings that can be canonicalized.  Therefore, iterate the
+      // simplification steps until converged.
+      while (true) {
+        auto start_of_loop = next;
+        next = Downcast<SeqExpr>(CanonicalizeBindings(next));
+        next = Downcast<SeqExpr>(EliminateCommonSubexpr(next));
+        next = Downcast<SeqExpr>(RemoveAllUnused(next));
+        if (struct_equal(start_of_loop, next)) {
+          break;
+        }
+      }
+
+      if (struct_equal(prev, next)) {
+        return std::move(next);
+      }
+
+      // Reset all knowledge of bindings that were collected from
+      // this SeqExpr.  The collected bindings are only after
+      // the point where they were collected, and we are repeating
+      // the mutation of this SeqExpr.
+      bindings_ = cache;
+      prev = next;
+    }
+  }
+
+  void VisitBinding_(const VarBindingNode* binding) override {
+    auto expr = VisitExpr(binding->value);
+    bindings_.Set(binding->var, expr);
+    ReEmitBinding(binding, expr);
+  }
+
+  Expr VisitExpr(const Expr& expr) override {
+    auto node = ExprMutator::VisitExpr(expr);
+
+    std::vector<DFPattern> matches_top_level;
+    if (auto rewritten = TryRewrite(node, pattern_, &matches_top_level)) {
+      return builder_->Normalize(rewritten.value());
+    }
+
+    return node;
+  }
+
+ private:
+  Optional<Expr> TryRewrite(const Expr& expr, const DFPattern& pattern,
+                            std::vector<DFPattern>* matches_top_level) {
+    ICHECK(matches_top_level);
+
+    // Special handling if the user-supplied pattern is a `OrPattern`.
+    // While the `ExtractMatchedExpr` can handle match the
```

Comment:
Looks like a typo. I assume it's supposed to be "handle matching," correct?

Reply:
Whoops, this was a typo.  It should be "handle matching", and I've updated the PR with the correction.